cluster/sessions.go

```diff
@@ -0,0 +1,101 @@
+// This file is part of MinIO Kubernetes Cloud
+// Copyright (c) 2019 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+package cluster
+
+import (
+	"database/sql"
+	"fmt"
+	_ "github.com/lib/pq"
+	"log"
+	"os"
+	"sync"
+)
+
+type singleton struct {
+	Db *sql.DB
+}
+
+var instance *singleton
+var once sync.Once
+
+// Returns a singleton instance that keeps the connections to the Database
+func GetInstance() *singleton {
+	once.Do(func() {
+		// Wait for the DB connection
+		db := <-ConnectToDb()
+
+		instance = &singleton{
+			Db: db,
+		}
+	})
+	return instance
+}
+
+func ConnectToDb() chan *sql.DB {
```

Comment:
This function should take a context (for cancellation) and return a `*sql.DB`. For example the current code would block forever in case of an error.
Maybe we can use something like:
```
ConnectToDb(ctx context.Context) (*sql.DB, error) {
      select {
            case // try to connect to DB
            case // abort when context is canceled 
      }
}
```

Reply:
This code has a fatal clause, since if there's not DB we wouldn't be able to do much, but yeah, i want to start using the context around to control cancellations, node that I `defer close(ch)` at the beginning of the function to avoid the block. I added the select clause.